pkg/apis/flyteworkflow/v1alpha1/nodes.go

```diff
@@ -93,17 +93,18 @@ func (in *NodeMetadata) DeepCopyInto(out *NodeMetadata) {
 }
 
 type NodeSpec struct {
-	ID            NodeID                        `json:"id"`
-	Name          string                        `json:"name,omitempty"`
-	Resources     *typesv1.ResourceRequirements `json:"resources,omitempty"`
-	Kind          NodeKind                      `json:"kind"`
-	BranchNode    *BranchNodeSpec               `json:"branch,omitempty"`
-	TaskRef       *TaskID                       `json:"task,omitempty"`
-	WorkflowNode  *WorkflowNodeSpec             `json:"workflow,omitempty"`
-	InputBindings []*Binding                    `json:"inputBindings,omitempty"`
-	Config        *typesv1.ConfigMap            `json:"config,omitempty"`
-	RetryStrategy *RetryStrategy                `json:"retry,omitempty"`
-	OutputAliases []Alias                       `json:"outputAlias,omitempty"`
+	ID                NodeID                        `json:"id"`
+	Name              string                        `json:"name,omitempty"`
+	Resources         *typesv1.ResourceRequirements `json:"resources,omitempty"`
+	ResourceOverrides *typesv1.ResourceRequirements `json:"resourceOverrides,omitempty"`
```

Comment:
this is getting complicated. Earlier we would compute the overrides in the compiler and send it here. Now we have both?

Reply:
sure, as an alternative we could compute resources and also keep track of platform defaults/limits and apply as needed in plugins